### PR TITLE
Update Redux store to display books 🤖 

### PR DESCRIPTION
**Pull request by evansnyamekye**
### Update Redux store to display books 📖. 

This pull request updates the bookstore to use Redux in React components. Redux is a state management library that helps to keep the state of the application. 

**The following modifications/changes were applied:** 

- Added a store variable to the index.js file 📑. 
- Added connect and mapStateProps function to the Bookstore component ⏱️.
- Updated the home page component to use the store variable and the mapStateToProps function ⚛️.  

Thank you 😃.

### CODE REVIEW

File: src/components/AddBook.jsx [hunk 1]
```diff
@@ -1,14 +1,45 @@
 /* eslint-disable import/no-extraneous-dependencies */
 import { v4 as uuidv4 } from 'uuid';
-import PropTypes from 'prop-types';
 import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addNewBook } from '../redux/books/bookSlice';
+
+function AddBook() {
```
Comment (TedLivist) [inline src/components/AddBook.jsx:6]: - [ ] Kindly look through your app. I cannot add a book. Regardless of what I submit, the book always defaults to Hunger Games. Kindly ensure the added books are reflected on the page.

File: src/components/BookList.jsx [hunk 2]
```diff
@@ -1,60 +1,35 @@
...
-          completed={todo.completed}
-          chapter={todo.chapter}
-          deleteBook={() => handleDelete(todo.id)}
+          key={item.id}
+          id={item.id}
+          title={item.title}
+          author={item.author}
+          category={item.category}
         />
       ))}
```
Comment (TedLivist) [inline src/components/BookList.jsx:28]: - [ ] I also cannot remove any book from the page, whether it is the initial book or the one added. The remove functionality does not at all. Kindly fix that.

File: src/components/AddBook.jsx [hunk 3]
```diff
@@ -33,17 +64,16 @@ function AddBook({ handleAdd }) {
 
   const handleSubmit = (e) => {
     e.preventDefault();
+    // const newId = uuidv4();
```
Comment (TedLivist) [inline src/components/AddBook.jsx]: - [ ] Kindly remove commented code.

File: src/components/BookList.jsx [hunk 4]
```diff
@@ -1,60 +1,35 @@
-import { useState } from 'react';
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (TedLivist) [inline src/components/BookList.jsx]: - [ ] Kindly avoid disabling linters as we need them to write clean code and maintain a standard

File: src/redux/Store.js [hunk 5]
```diff
@@ -1,12 +1,12 @@
 /* eslint-disable import/no-extraneous-dependencies */
```
Comment (TedLivist) [inline src/redux/Store.js]: - [ ] Kindly remove linter disablings and fix the resulting linter error manually.

File: src/redux/books/bookSlice.js [hunk 6]
```diff
@@ -2,24 +2,45 @@
...
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      item_id: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
```
Comment (TedLivist) [inline src/redux/books/bookSlice.js]: - [ ] These initial books are not listed on the page when I load the page. Kindly fix that.

  image

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @evansnyamekye,

Good job so far!
✔️ Provider wrapper for app in `index.js` file
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @evansnyamekye 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good project organization ✔️
- [x] Nice use of React features ✔️  
- [x] Kudus on meeting the requirements as  required  ✔️


Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**